vanilla-tests/src/main/java/fixtures/clientflattendiscriminatorflattening/models/BackendAddressPool.java

```diff
@@ -0,0 +1,72 @@
+package fixtures.clientflattendiscriminatorflattening.models;
+
+import com.azure.core.annotation.Fluent;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/** Pool of backend IP addresses. */
+@Fluent
+public final class BackendAddressPool {
+    /*
+     * Properties of load balancer backend address pool.
+     */
+    @JsonProperty(value = "properties")
+    private BackendAddressPoolPropertiesFormat properties;
+
+    /**
+     * Get the properties property: Properties of load balancer backend address pool.
+     *
+     * @return the properties value.
+     */
+    private BackendAddressPoolPropertiesFormat getProperties() {
+        return this.properties;
+    }
+
+    /**
+     * Set the properties property: Properties of load balancer backend address pool.
+     *
+     * @param properties the properties value to set.
+     * @return the BackendAddressPool object itself.
+     */
+    private BackendAddressPool setProperties(BackendAddressPoolPropertiesFormat properties) {
+        this.properties = properties;
+        return this;
+    }
+
+    /**
+     * Get the location property: The location of the backend address pool.
+     *
+     * @return the location value.
+     */
+    public String getLocation() {
+        if (this.getProperties() == null) {
+            return null;
+        } else {
+            return this.getProperties().getLocation();
+        }
+    }
+
+    /**
+     * Set the location property: The location of the backend address pool.
+     *
+     * @param location the location value to set.
+     * @return the BackendAddressPool object itself.
+     */
+    public BackendAddressPool setLocation(String location) {
+        if (this.getProperties() == null) {
+            this.setProperties(new BackendAddressPoolPropertiesFormat());
+        }
+        this.getProperties().setLocation(location);
+        return this;
+    }
```

Comment:
This one is not thread safe. We could have race condition here.

Any suggestion? A `syncrhonized` `ensureProperties` method?